test/java/seedu/addressbook/logic/LogicTest.java

```diff
@@ -90,24 +90,26 @@ private void assertCommandBehavior(String inputCommand,
         //Confirm the state of data is as expected
         assertEquals(expectedAddressBook, addressBook);
         assertEquals(lastShownList, logic.getLastShownList());
-        assertEquals(addressBook, saveFile.load());
+
+        // verify if the savFile matches address book if and only if the command mutates the data
+        if(hasMutated){assertEquals(addressBook, saveFile.load());}
```

Comment:
Does this check the only if condition? i.e. If the command that is NOT mutating it should check if it doesn't save the file.